### ISSUE TITLE
[Bug] Crash when opening directory on Arch involving FromUnixTimeSeconds

### ISSUE THREAD

**Issue by waterlubber** [KSP-CKAN/CKAN#3473]
Background
----------
* Operating System: Arch Linux, mono 6.12.0
* CKAN Version: v1.30.4
* KSP Version: N/A

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?**
N/A

Problem
-------
When running CKAN and selecting the parent directory of my KSP installs, it immediately crashes with the following error:
```3843 [4] ERROR CKAN.ErrorDialog (null) - Unhandled exception:
System.ArgumentOutOfRangeException: Valid values are between -62135596800 and 253402300799, inclusive.
Parameter name: seconds
  at System.DateTimeOffset.FromUnixTimeSeconds (System.Int64 seconds) [0x00043] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileStatus.UnixTimeToDateTimeOffset (System.Int64 seconds, System.Int64 nanoseconds) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileStatus.GetLastAccessTime (System.ReadOnlySpan`1[T] path, System.Boolean continueOnError) [0x00018] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTimeCore () [0x00011] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTimeUtc () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTime () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemInfo.get_LastAccessTime()
  at System.Windows.Forms.FileSystem.GetDirectoryFSEntry (System.IO.DirectoryInfo dirinfo, System.Windows.Forms.FSEntry topFolderFSEntry) [0x00047] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.FileSystem.GetNormalFolderContent (System.String from_folder, System.Collections.Specialized.StringCollection filters, System.Collections.ArrayList& directories_out, System.Collections.ArrayList& files_out) [0x00023] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.FileSystem.GetFolderContent (System.Collections.Specialized.StringCollection filters, System.Collections.ArrayList& directories_out, System.Collections.ArrayList& files_out) [0x00155] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.MWFVFS+WorkerThread.GetFolderContentThread () [0x00000] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <62b430b945fa49a19a75382ef03e7bed>:0  
```

**Steps to reproduce**
* Run CKAN via mono ckan.exe
* Select New Game Instance -> Add Instance to CKAN
* Navigate to a folder
* CKAN crashes

**Expected behavior**
CKAN does not crash and instead correctly detects the instance. Notably, I have maintained this same folder of KSP installs for many years and it worked with my prior CKAN versions, the issue has manifested since I last played. It also occurs with the ckan-bin package on the AUR.

**Screenshots (if applicable)**
![image](https://user-images.githubusercontent.com/7517025/140663421-b3639983-a661-4700-9122-980b1fcaf49b.png)

**Comment by HebaruSan** [comment]
Hmm, there's no CKAN code in that call stack (it's just the standard file/folder browser trying to get info from the file system), so this is probably a problem with your Mono install.

What's special about the parent directory of your KSP installs that might lead to it having a weird timestamp?

**Comment by waterlubber** [comment]
Absolutely nothing, as far as I can tell (that was the first thing I checked):
```
$ ls -hal
drwxr--r--   6 waterlubber waterlubber 3.5K Nov  7 16:56  .
drwxr-xr-x 210 waterlubber waterlubber  52K Nov  7 19:05  ..
-rw-r--r--   1 waterlubber waterlubber 3.4M Nov  7 16:55  ckan.exe
drwxr-xr-x  12 waterlubber waterlubber 3.5K Jul 21  2020  KSP_10
drwxr-xr-x  12 waterlubber waterlubber 3.5K Jul 28 17:54  KSP_linux
drwxrwxr-x   9 waterlubber waterlubber 3.5K Aug 28  2020 'KSP Saves'
drwxrwxr-x   4 waterlubber waterlubber 3.5K Aug 28  2020  Screenshots
```
I went ahead and reinstalled mono from the Arch repositories, but no luck there.

I also ran `find . -exec stat -c "%Y" {} \; | uniq | sort > times.txt` to try and search for unusual timestamps, but all of the results were between `1412811164` and `1636330253` -- all reasonable.

I *did* find some really unusual values when running %X - checking for *access time*, instead of modification time. These were in a freshly extracted zip of ksp_linux (1.12.2), so it's possible that zip extraction scrambled them.

After clearing these with touch, the problem went away - so somewhere in the zip extraction process for KSP there was an error that messed up the timestamps.

Not sure to consider this a bug of mono or just a weird issue on my hand - if this is a semi-common issue when extracting KSP archives, it might be worth it to add handling.
